core/src/main/java/org/lflang/target/property/PlatformProperty.java

```diff
@@ -75,10 +75,10 @@ protected PlatformOptions fromString(String string, MessageReporter reporter) {
   @Override
   public void validate(TargetConfig config, MessageReporter reporter) {
     var singleThreaded = config.get(SingleThreadedProperty.INSTANCE);
-    if (!singleThreaded && config.get(PlatformProperty.INSTANCE).platform == Platform.RP2040) {
+    if (!singleThreaded && !config.get(PlatformProperty.INSTANCE).platform.isMultiThreaded()) {
```

Comment:
I'm guessing that this check was meant to be more general than just the RP2040, but was introduced before the `platform.isMultiThreaded` attribute. Currently all platforms are multithreaded so this check isn't very useful yet.